Description: Mark distribution search plugins as read only
 This brings the behaviour of SearchService.prototype._asyncLoadEnginesFromDir()
 in line with the sync code path (SearchService.prototype._loadEnginesFromDir).
 Without  this, if a user tries to remove one of our custom search engines, it is
 deleted from the search engine cache rather than marked as hidden. This results
 in it reappearing on cache rebuilds (eg, on app upgrades).
 Note that SearchService.prototype._asyncLoadEnginesFromDir() is only called on
 distribution directories, so they are always read only
Author: Chris Coulson <chris.coulson@canonical.com>
Forwarded: no
Bug-Ubuntu: https://launchpad.net/bugs/1791789

--- a/toolkit/components/search/nsSearchService.js
+++ b/toolkit/components/search/nsSearchService.js
@@ -3267,7 +3267,7 @@ SearchService.prototype = {
       try {
         let file = Cc["@mozilla.org/file/local;1"].createInstance(Ci.nsIFile);
         file.initWithPath(osfile.path);
-        addedEngine = new Engine(file, false);
+        addedEngine = new Engine(file, true);
         await checkForSyncCompletion(addedEngine._asyncInitFromFile(file));
         engines.push(addedEngine);
       } catch (ex) {
